src/Controller/ApiV1Controller.php

```diff
@@ -0,0 +1,211 @@
+<?php
+/**
+ * Tatoeba Project, free collaborative creation of multilingual corpuses project
+ * Copyright (C) 2009-2010  HO Ngoc Phuong Trang <tranglich@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  Tatoeba
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+namespace App\Controller;
+
+use App\Controller\AppController;
+use App\Form\SentencesSearchForm;
+use App\Model\CurrentUser;
+use App\Model\Table\SentencesTable;
+use App\Lib\LanguagesLib;
+use App\Lib\SphinxClient;
+use App\Lib\Licenses;
+use Cake\Core\Configure;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Event\Event;
+use Cake\Routing\Router;
+use Cake\Utility\Hash;
+use Cake\View\ViewBuilder;
+use Exception;
+
+/**
+ * Controller for Api.
+ *
+ * @category Sentences
+ * @package  Controllers
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+class ApiV1Controller extends AppController
+{
+    public $name = 'Sentences';
```

Comment:
I think this is not the correct name for this controller. You can simply remove this line and CakePHP will derive the name from the class name.

Reply:
OK.

Reply:
In theory that sounds like it makes sense.

In practice, commenting it out breaks everything:
![image](https://user-images.githubusercontent.com/7943077/117715168-367ac580-b1d8-11eb-8ce2-92124d90e0ba.png)

with the referenced line 93:
![image](https://user-images.githubusercontent.com/7943077/117715428-88235000-b1d8-11eb-8eda-ebdc87db1f15.png)

![image](https://user-images.githubusercontent.com/7943077/117715465-970a0280-b1d8-11eb-94a0-f1c96eada45f.png)

Line 143 looks a bit weird:
![image](https://user-images.githubusercontent.com/7943077/117715610-cae52800-b1d8-11eb-9732-5dd1fe6d0bd2.png)

I'm not sure why it breaks things commenting it out, but it works with it in.

Reply:
Do you remember when you had to insert some `$this->loadModel()` after copy-pasting code from a different controller? It’s a similar problem here.

Until now, the controller was named "Sentences", so the model "Sentences" was automatically loaded by CakePHP. Therefore, your `sentence()` method could access the model with `$this->Sentences->...` directly. However, now that you renamed the controller, the "Sentences" is not automatically loaded any more, so you need to `loadModel()` it. If there was an "ApiV1" model, CakePHP would load if for you, but obviously there is no such model, so no model is automatically loaded.

Reply:
OK, I've removed the "public $name = 'Sentences';" line and added loadModel('Sentences') to both the sentence() and search() methods:
![image](https://user-images.githubusercontent.com/7943077/118846615-31e79880-b8cd-11eb-9398-e0153fdcc936.png)